XSConsoleTask.py

```diff
@@ -53,11 +56,7 @@ def Status(self):
         return status
 
     def Result(self):
-        if self.Completed():
-            result = self.completionStatus
-        else:
-            result= self.session.xenapi.task.get_status(self.hotOpaqueRef.OpaqueRef())
-        return HotOpaqueRef(result, 'any')
+        return self.result
```

Comment:
For synchronous tasks, is it guaranteed that `self.result` has already been set?

Reply:
The caller of `Result()` checks if the result is none, and raises an exception if none.